addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/discovery/FreeboxDiscoveryService.java

```diff
@@ -0,0 +1,100 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.freebox.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.matmaul.freeboxos.FreeboxException;
+import org.matmaul.freeboxos.FreeboxOsClient;
+import org.matmaul.freeboxos.login.Authorize;
+import org.matmaul.freeboxos.login.LoginManager;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.openhab.binding.freebox.FreeboxBindingConstants.*;
+
+/**
+ * The {@link FreeboxDiscoveryService} is responsible for discovering a Freebox Thing
+ * on the local network
+ * 
+ * @author Gaël L'hopital - Initial contribution
+ */
+
+public class FreeboxDiscoveryService extends AbstractDiscoveryService {
+
+	private static final Logger logger = LoggerFactory.getLogger(FreeboxDiscoveryService.class);
+
+	public FreeboxDiscoveryService() {
+		super(SUPPORTED_THING_TYPES_UIDS,10);
+	}
+
+	public Set<ThingTypeUID> getSupportedThingTypes() {
+		return SUPPORTED_THING_TYPES_UIDS;
+	}
+	
+	@Override
+	protected void startScan() {
+		discoverFreebox();
+	}
+
+
+	/**
+	 * Discover the Freebox, get its appToken then submits the result
+	 */
+	private void discoverFreebox(){
+		Bundle bundle = FrameworkUtil.getBundle(getClass());			
+		
+		FreeboxOsClient fbClient = new FreeboxOsClient(
+					bundle.getSymbolicName() /*  org.openhab.binding.freebox */
+					);
+		
+		LoginManager loginManager = fbClient.getLoginManager();
+		
+		try {
+			Authorize authorize = loginManager.newAuthorize(
+					bundle.getHeaders().get("Bundle-Name"),													// Freebox Binding
+					String.format("%d.%d",bundle.getVersion().getMajor(),bundle.getVersion().getMinor()), 	// eg. 1.5
+					bundle.getHeaders().get("Bundle-Vendor"));
+			String appToken = authorize.getAppToken();
+
+			if (appToken !=null){
+				submitDiscoveryResults("server", appToken);}
+		} catch (FreeboxException e) {
+			logger.error(e.getMessage());
+		}								
+	}
+
+	/**
+	 * Submit the discovered location to the Smarthome inbox
+	 * @param thingName
+	 * @param appToken
+	 */
+	private void submitDiscoveryResults(String thingName, String appToken) {
+		ThingUID uid = new ThingUID(FREEBOX_THING_TYPE_SERVER, thingName );
+		if(uid!=null) {
+			Map<String, Object> properties = new HashMap<>(1);
+			properties.put("appToken" ,appToken);
+			DiscoveryResult result = DiscoveryResultBuilder.create(uid)
+					.withProperties(properties)
+					.withLabel("Freebox Revolution " + thingName)
```

Comment:
Free provides a second appliance (a media server), I kept room for introducing this one later but will clean the code.